bin/templates/cordova/Api.js

```diff
@@ -0,0 +1,506 @@
+/**
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+*/
+
+var Q = require('q');
+var fs = require('fs');
+var path = require('path');
+var shell = require('shelljs');
+
+var CordovaError = require('cordova-common').CordovaError;
+var PlatformJson = require('cordova-common').PlatformJson;
+var ActionStack = require('cordova-common').ActionStack;
+var AndroidProject = require('./lib/AndroidProject');
+var PlatformMunger = require('cordova-common').ConfigChanges.PlatformMunger;
+var PluginInfoProvider = require('cordova-common').PluginInfoProvider;
+
+var pluginHandlers = require('./lib/pluginHandlers');
+
+var PLATFORM = 'android';
+var GENERIC_EVENTS = new (require('events').EventEmitter)()
+    .on('verbose', function (message) {
+        if (process.argv.indexOf('-d') >= 0 || process.argv.indexOf('--verbose') >= 0)
+            console.log(message);
+    })
+    .on('log', console.log)
+    .on('warn', console.warn)
+    .on('error', function (error) {
+        if (process.argv.indexOf('-d') >= 0 || process.argv.indexOf('--verbose') >= 0) {
+            console.error((error && error.message) || error);

```

Comment:
> Should this be flipped? We should stack for verbose output and not otherwise?

Yes, definitely. My bad.

> Also, does the message not show when we show stack?

The message is shown along with stack, yes. Just to clarify, the idea here is that until you don't have `-d` switch, you won't get full stacks, because they are only interesting for those, who working with cordova-lib/platform internals. Common users should see only meaningful error message , that explains what went wrong. This logic is borrowed [from CLI ](https://github.com/apache/cordova-cli/blob/c5acc3dcbea7ac736f778e58cde23a38b467572a/src/cli.js#L130)

>  If I understand correctly, this is only for legacy scenarios - correct? Otherwise, the cli should eventually be doing the logging - correct?

This code is required for cases, when caller of PlatformApi doesn't provide logger/eventEmitter instance. In this case we still need to log some events. In particular, this code is used in non-CLI workflow. For CLI workflow - yes, logging is already handled by CLI.